flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/comparison.scala

```diff
@@ -74,6 +109,8 @@ case class IsNull(child: Expression) extends UnaryExpression {
   override def toRexNode(implicit relBuilder: RelBuilder): RexNode = {
     relBuilder.isNull(child.toRexNode)
   }
+
+  override def resultType = BOOLEAN_TYPE_INFO
```

Comment:
`IsNull` accepts all type of columns and returns if the current cell is `null`, therefore, the `resultType` is `Boolean`. `IsNull` is extending `UnaryExpression` which does not have a default `resultType` implementation, so we need override.